File: 560.subarray-sum-equals-k.go
func subarraySum(nums []int, k int) int {
    total := 0
    for i := 1; i < len(nums); i++ {
        nums[i] += nums[i-1]
    }
    for i := 0; i < len(nums); i++ {
        if nums[i] == k {
            total++
        }
        for j := 0; j < i; j++ {
            if nums[i] - nums[j] == k {
                total++
            }
        }
    }
    return total
}

File: 560.subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        preCnt = {0: 1}
        res = 0
        pre = [0] * len(nums)
        for i in range(len(nums)):
            pre[i] = nums[i]

            if i > 0:
                pre[i] += pre[i - 1]
            # print('preCnt: ', preCnt, ' pre[i]', pre[i])
            res += preCnt.get(pre[i] - k, 0)
            preCnt[pre[i]] = preCnt.get(pre[i], 0) + 1
        return res
